### DISCUSSION TITLE
Sync Night Shift and True Tone to external monitor

### DISCUSSION THREAD

**Discussion by patrickrushton** [MonitorControl/MonitorControl#911] [General]
### Before opening the issue, have you...?

- [X] Searched for existing issues

### Is your feature request related to a problem? Please describe

Generally my Mac will sync the True Tone and Night Shift to my external monitor. Except when waking from sleep or coming out of clamshell mode, only the integrated display will have warmer colours while the external monitor resets to standard, relatively cooler colours.

### Describe the solution you'd like

Maybe this is something MonitorControl can help with. For example by forcing the external display to sync the colour tinting to the True Tone and Night Shift values of the integrated display.

### Describe alternatives you've considered

One alternative is to disable and reenable Night Shift / True Tone in System Preferences whenever the computer wakes up in the evening. That normally forces the external display to update its colour tint.

Another option is to power cycle the external (Dell) display.

**Comment by waydabber** [comment]
Well, I am afraid MC will not be able to help with this. MC can alter the colors of the screen and create the same effect as Night Shift, but Night Shift and True Tone are actually stacked above this by one level and is applied on the color collercted image that would be produced by MC. Since this seems to be a macOS bug, I'd report this one to Apple, hoping for a fix.

**Comment by waydabber** [comment]
(I'll convert this one to a discussion, we can continue it there).

**Reply by will-huang9531** [reply]
Hey man
One of the features of MacOS is that if the external display is recognized as a TV or projector, True Tone and Night Shift will not be applied to the external display. Many monitors (especially DIY) are recognized as 61 inch TVs, so they cannot use these two functions.
In the past, the EDID of the external display can be replaced by overrides in the Intel era, so that it can be disguised as an Apple display, but this is no longer effective in the apple silicon era. This has caused great trouble to M1 users.
As we all know, Monitor control is a famous and effective open source project, which has saved many Mac users. Therefore, can we consider adding such a function to enable the external display to turn on True Tone and Night Shift, or if it is not technically feasible, can we enable the external display to load a special filter separately？

**Comment by vurtomatic** [comment]
Here for this too.

**Comment by jonsedar** [comment]
+1 for a filter please to alter white balance (similar to True Tone)

I'm using an M2 MacBook Air attached to an LG Ultrafine 5K. With the lid open, the MacBook light sensor automatically adjusts the white balance on the laptop and monitor. With the lid closed, the monitor reverts to default blueish white light. 

There doesn't seem to be an Apple-y way to change the white balance, so I'd love to have a filter in Monitor Control please :)
